WorkloadManagementSystem/Client/Matcher.py

```diff
@@ -276,10 +276,14 @@ def _updatePilotJobMapping( self, resourceDict, jobID ):
   def _checkCredentials( self, resourceDict, credDict ):
     """ Check if we can get a job given the passed credentials
     """
-    # Check credentials if not generic pilot
     if Properties.GENERIC_PILOT in credDict[ 'properties' ]:
       # You can only match groups in the same VO
-      vo = Registry.getVOForGroup( credDict[ 'group' ] )
+      if credDict[ 'group' ] == "hosts":
+        # for the host case the VirtualOrganization parameter
+        # is mandatory in resourceDict
+        vo = resourceDict.get( 'VirtualOrganization', '' )
```

Comment:
In multi-VO installations it will finally result in an exception and the jobs will not be matched